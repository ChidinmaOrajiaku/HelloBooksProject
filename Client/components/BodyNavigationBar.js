import React from 'react';

/**
 * 
 * 
 * @class BodyNavigationBar
 * @extends {React.Component}
 */
class BodyNavigationBar extends React.Component {
  /**
   * 
   * 
   * @returns 
   * @memberof BodyNavigationBar
   */
  render() {
    $(document).ready(() => {
    $(".button-collapse").sideNav();
  });
    return (
      <div className="container-fluid navbar-fixed">
        <nav>
          <div className="nav-wrapper teal">
            <a href="#!" className="brand-logo">HelloBooks</a>
            <a href="#" data-activates="mobile-demo" className="button-collapse"><i className="material-icons">menu</i></a>
            <ul className="right hide-on-med-and-down">
              <li><a href="/register">Sign Up</a></li>
              <li><a href="/login">Sign In</a></li>
            </ul>
            <ul className="side-nav" id="mobile-demo">
            <li><a href="/register">Sign Up</a></li>
            <li><a href="/login">Sign In</a></li>
            </ul>
          </div>
        </nav>

      </div>
    );
  }
}

export default (BodyNavigationBar);
