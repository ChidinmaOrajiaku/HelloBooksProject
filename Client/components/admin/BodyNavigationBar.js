import React from 'react';


class BodyNavigationBar extends React.Component {
  render(){
  $(document).ready(function(){
    $(".button-collapse").sideNav();
  })
  return (
      <div className="container-fluid navbar-fixed">
         <nav>
         <div className="nav-wrapper teal">
           <a href="#!" className="brand-logo">HelloBooks</a>
           <a href="#" data-activates="mobile-demo" className="button-collapse"><i className="material-icons">menu</i></a>
           <ul className="right hide-on-med-and-down">
             <li><a href="/library">Library</a></li>
             <li><a href="/history">History</a></li>
             <li><a href="/profile">Profile</a></li>
             <li><a href="/">Log Out</a></li>
           </ul>
           <ul className="side-nav" id="mobile-demo">
           <li><a href="/library">Library</a></li>
           <li><a href="/history">History</a></li>
           <li><a href="/profile">Profile</a></li>
           <li><a href="/">Log Out</a></li>
           </ul>
         </div>
       </nav>
               
      </div>
   );
}
};

export default (BodyNavigationBar);
