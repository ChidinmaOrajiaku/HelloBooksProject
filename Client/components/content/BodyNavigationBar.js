import React from 'react';


export default () => {
        $(".button-collapse").sideNav();
  return (
      <div className="container-fluid">
         <nav>
         <div className="nav-wrapper teal">
           <a href="#!" className="brand-logo">Logo</a>
           <a href="#" data-activates="mobile-demo" className="button-collapse"><i className="material-icons">menu</i></a>
           <ul className="right hide-on-med-and-down">
             <li><a href="/library">Library</a></li>
             <li><a href="collapsible.html">History</a></li>
             <li><a href="mobile.html">Profile</a></li>
           </ul>
           <ul className="side-nav" id="mobile-demo">
             <li><a href="badges.html">Library</a></li>
             <li><a href="collapsible.html">History</a></li>
             <li><a href="mobile.html">Profile</a></li>
           </ul>
         </div>
       </nav>
               
      </div>
   );
};
