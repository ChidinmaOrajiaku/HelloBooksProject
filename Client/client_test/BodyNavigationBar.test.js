import React from 'react';
import $ from 'jquery';
import toJson from 'enzyme-to-json';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import  BodyNavigationBar  from '../components/BodyNavigationBar.jsx';

configure({ adapter: new Adapter() });

describe('<BodyNavigationBar />', () => {
  it('should render component without breaking', () => {
    const wrapper = shallow(<BodyNavigationBar />);
    expect(wrapper).toBeDefined();
  });
  it('should match snapshot test', () => {
    const component = shallow(<BodyNavigationBar />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
