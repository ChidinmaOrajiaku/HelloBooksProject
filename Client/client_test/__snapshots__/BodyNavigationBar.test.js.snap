// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BodyNavigationBar /> should match snapshot test 1`] = `
<div
  className="container-fluid navbar-fixed"
>
  <nav>
    <div
      className="nav-wrapper teal"
    >
      <Link
        className="brand-logo"
        replace={false}
        to="/"
      >
        HelloBooks
      </Link>
      <ul
        className="right hide-on-med-and-down"
      >
        <li>
          <Link
            replace={false}
            to="/register"
          >
            Sign Up
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/login"
          >
            Sign In
          </Link>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
